sqlx-sqlite/src/connection/mod.rs

```diff
@@ -492,6 +503,34 @@ impl LockedSqliteHandle<'_> {
     pub fn remove_rollback_hook(&mut self) {
         self.guard.remove_rollback_hook();
     }
+
+    pub(crate) fn in_transaction(&mut self) -> bool {
+        let ret = unsafe { sqlite3_get_autocommit(self.as_raw_handle().as_ptr()) };
+        ret == 0
+    }
+
+    /// Calls `sqlite3_txn_state` on this handle.
+    pub fn transaction_state(&mut self) -> Result<SqliteTransactionState, Error> {
+        use libsqlite3_sys::{
+            sqlite3_txn_state, SQLITE_TXN_NONE, SQLITE_TXN_READ, SQLITE_TXN_WRITE,
+        };
+
+        let state =
+            match unsafe { sqlite3_txn_state(self.as_raw_handle().as_ptr(), std::ptr::null()) } {
+                SQLITE_TXN_NONE => SqliteTransactionState::None,
+                SQLITE_TXN_READ => SqliteTransactionState::Read,
+                SQLITE_TXN_WRITE => SqliteTransactionState::Write,
+                _ => return Err(Error::Protocol("Invalid transaction state".into())),
+            };
+        Ok(state)
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SqliteTransactionState {
+    None,
+    Read,
+    Write,
```

Comment:
Not sure about exposing this at this time. [No one has ever asked for it](https://github.com/search?q=repo%3Alaunchbadge%2Fsqlx+sqlite3_txn_state&type=issues) and I'd want to workshop the API a bit which would be out-of-scope for this PR.

Reply:
Fair, though I'm not sure how I should hide it, since it's used by a different crate.

Reply:
It might be a good idea to coordinate this PR with https://github.com/launchbadge/sqlx/pull/3427 then, since we're already exposing a way to get the depth in that one.

Reply:
I think maybe I'll just push this method all the way out into the test then. I don't really want to be coupled to another PR if I can help it.